icechunk/src/session.rs

```diff
@@ -240,6 +242,26 @@ impl Session {
             .map(|node| self.change_set.update_array(node.id, metadata))
     }
 
+    pub async fn array_has_modified_chunks(
+        &self,
+        path: Path,
+        id: NodeId,
+    ) -> SessionResult<bool> {
+        let array = self.get_array(&path).await?;
+        if let NodeSnapshot { node_data: NodeData::Array(_, manifests), .. } = array {
+            debug_assert_eq!(array.id, id);
+            let path_and_id = &(path, id);
+            Ok(!self.change_set.array_is_deleted(path_and_id)
+                && (self.change_set.array_has_modified_chunks(path_and_id)
+                    || !manifests.is_empty()))
```

Comment:
I think this misses the case of a session that deletes all chunks in an existing array. But see my comment below, maybe we are doing too much... maybe we can always return the `/c`

Reply:
yes... i had the same realization in https://github.com/earth-mover/icechunk/pull/528#discussion_r1900282037.

I've updated to always return `c`. We can get smarter if someone complains and really cares about it.